### PR TITLE
Extend flags of `manifest add`

### PR DESCRIPTION
**Pull request by flavio**
#### What type of PR is this?

/kind cleanup
/kind feature

#### What this PR does / why we need it:

This PR extends the flags of `buildah manifest add` to include also:

  * cert-dir
  * auth-file
  * creds
  * tls-verify

These options are useful when adding to a manifest an image that is not part of the local image store. The image resides on a remote registry that falls into one of these cases: it's not using tls termination, it requires authentication or it's secured with an unknown tls certificate.

Consider the following scenario: a multi architecture manifest is created as part of a multi-step CI pipeline running in a containerized way.
All the images referenced by the manifest live inside of a registry secured with a self-signed tls certificate.

Without this patch the manifest creation step would have to pull all the multi-architecture images locally via `buildah pull`.

With this patch the usage of `buildah pull` would not be needed because the images' digests can be requested straight to the registry. That means the execution of manifest creation step would be faster and result in less disk space and network bandwidth being used.

#### How to verify it

Prepare the following environments:

* Insecure container registry: `insecure-reg.local.lan`
* Container registry secured by self-signed ceriticate: `self-signed-reg.local.lan`
* Container registry with authentication enforced, even for pull options: `auth-reg.local.lan`

Ensure each repository has at least an image inside of it: 
  * `<registry>/busybox:latest-amd64`
  * `<registry>/busybox:latest-arm64`

On a machine where no container images are defined execute the following steps:

* Create a manifest for the insecure registry:

```
buildah manifest create insecure-reg.local.lan/busybox:latest
buildah manifest add --tls-verify=false insecure-reg.local.lan/busybox:latest docker://insecure-reg.local.lan/busybox:latest-amd64 docker://insecure-reg.local.lan/busybox:latest-arm64
buildah manifest push --tls-verify=false insecure-reg.local.lan/busybox:latest
```

* Create a manifest for the registry secured by a self-signed certificate not known by the host:

```
mkdir ~/certs
cp self-signed-reg.local.lan.crt ~/certs
buildah manifest create self-signed-reg.local.lan/busybox:latest
buildah manifest add --cert-dir=~/certs self-signed-reg.local.lan/busybox:latest docker://self-signed-reg.local.lan/busybox:latest-amd64 docker://self-signed-reg.local.lan/busybox:latest-arm64
buildah manifest push --cert-dir=~/certs self-signed-reg.local.lan/busybox:latest
```

* Create a manifest for the registry secured by a self-signed certificate and using authentication:

```
mkdir ~/certs
cp auth-reg.local.lan.crt ~/certs
buildah manifest create auth-reg.local.lan/busybox:latest
buildah manifest add --cert-dir=~/certs --creds alice:secret auth-reg.local.lan/busybox:latest docker://auth-reg.local.lan/busybox:latest-amd64 docker://auth-reg.local.lan/busybox:latest-arm64
buildah manifest push --cert-dir=~/certs  --creds alice:secret auth-reg.local.lan/busybox:latest
```

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

Nothing special to report.

#### Does this PR introduce a user-facing change?

```release-note
* Extend the `manifest add` command to support also `cert-dir`, `auth-file`, `creds` and `tls-verify` flags like other buildah command already do.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
Thanks @flavio 

Could you add command completions?

Once we have this merged, would you be interested do the same for podman?

**Comment by rhatdan** [comment]
@nalind @mtrmac @vrothberg @TomSweeneyRedHat PTAL

**Comment by rhatdan** [comment]
@QiWang19 PTAL

**Comment by flavio** [comment]
> Thanks @flavio
> 
> Could you add command completions?
> 
> Once we have this merged, would you be interested do the same for podman?

Sure, I'll take care about all of that.

**Comment by QiWang19** [comment]
LGTM with the command completion.

File: docs/buildah-manifest-add.md [hunk 1]
```diff
@@ -36,6 +36,22 @@ the image.  If *imageName* refers to a manifest list or image index, the
...
 
+**--authfile** *path*
+
+Path of the authentication file. Default is ${XDG\_RUNTIME\_DIR}/containers/auth.json, which is set using `buildah login`.
+If the authorization state is not found there, $HOME/.docker/config.json is checked, which is set using `docker login`.
+
+**--cert-dir** *path*
+
+Use certificates at *path* (\*.crt, \*.cert, \*.key) to connect to the registry.
+Default certificates directory is _/etc/containers/certs.d_.
```
Comment (TomSweeneyRedHat) [inline docs/buildah-manifest-add.md:47]:
```suggestion
The default certificates directory is _/etc/containers/certs.d_.
```

File: docs/buildah-manifest-add.md [hunk 2]
```diff
@@ -58,6 +74,10 @@ for the image.  This option is rarely used.
 Specify the OS version which the list or index records as a requirement for the
 image.  This option is rarely used.
 
+**--tls-verify** *bool-value*
+
+Require HTTPS and verify certificates when talking to container registries (defaults to true)
```
Comment (TomSweeneyRedHat) [inline docs/buildah-manifest-add.md:79]:
```suggestion
Require HTTPS and verify certificates when talking to container registries (defaults to true).
```

File: docs/buildah-manifest-add.md [hunk 3]
```diff
@@ -36,6 +36,22 @@ the image.  If *imageName* refers to a manifest list or image index, the
 architecture information will be retrieved from it.  Otherwise, it will be
 retrieved from the image's configuration information.
 
+**--authfile** *path*
+
+Path of the authentication file. Default is ${XDG\_RUNTIME\_DIR}/containers/auth.json, which is set using `buildah login`.
+If the authorization state is not found there, $HOME/.docker/config.json is checked, which is set using `docker login`.
```
Comment (TomSweeneyRedHat) [inline docs/buildah-manifest-add.md:42]: probably ought to add at least `buildah-login(1)` if not also `docker-login(1)` to the `See Also` section at the bottom of this man page.

**Comment by TomSweeneyRedHat** [comment]
code LGTM, would like a @nalind head nod.  Couple small doc/man page nits.

File: docs/buildah-manifest-add.md (hunk 1 above)
Comment (flavio) [reply docs/buildah-manifest-add.md:47]: I'm fine adding this suggestion, and the next one. Keep in mind I copied that from the `manifest push` man page.

If you don't mind I would ensure the style is consistent across all the man pages of buildah.

**Comment by flavio** [comment]
@TomSweeneyRedHat : I've updated the man pages of the `manifest add` and other ones to ensure they are all implementing your suggestions.

**Comment by rhatdan** [comment]
Something is going wrong in the tests?

**Comment by nalind** [comment]
LGTM
bors retry

**Comment by QiWang19** [comment]
bors retry

File: docs/buildah-manifest-add.md [hunk 4]
```diff
@@ -82,4 +102,4 @@ buildah manifest add --arch arm64 --variant v8 mylist:v1.11 docker://fedora@sha2
 ```
 
 ## SEE ALSO
-buildah(1), buildah-manifest(1), buildah-manifest-create(1), buildah-manifest-remove(1), buildah-manifest-annotate(1), buildah-manifest-inspect(1), buildah-manifest-push(1), buildah-rmi(1)
+buildah(1), buildah-login(1), buildah-manifest(1), buildah-manifest-create(1), buildah-manifest-remove(1), buildah-manifest-annotate(1), buildah-manifest-inspect(1), buildah-manifest-push(1), buildah-rmi(1),docker-login(1)
```
Comment (TomSweeneyRedHat) [inline docs/buildah-manifest-add.md:105]: nit of a nit, can you add a space before docker-login(1) please?
```suggestion
buildah(1), buildah-login(1), buildah-manifest(1), buildah-manifest-create(1), buildah-manifest-remove(1), buildah-manifest-annotate(1), buildah-manifest-inspect(1), buildah-manifest-push(1), buildah-rmi(1), docker-login(1)
```

**Comment by TomSweeneyRedHat** [comment]
@flavio ty for all the touch-ups.  I found one small nit.  Can you fix that and repush?  Sometimes bors likes PR's better after they're repushed for some reason.

File: docs/buildah-manifest-add.md (hunk 4 above)
Comment (flavio) [reply docs/buildah-manifest-add.md]: Good catch, I fixed that

**Comment by flavio** [comment]
> @flavio ty for all the touch-ups. I found one small nit. Can you fix that and repush? Sometimes bors likes PR's better after they're repushed for some reason.

Fixed the small nit and did a rebase to include the fix into one of the previous commits of the PR.

I just did a force push, let see how bors feels today ;)

**Comment by rhatdan** [comment]
bors retry

**Comment by TomSweeneyRedHat** [comment]
@flavio bors liked you!  Thanks!

**Pull request merged**